src/cfnlint/rules/mappings/KeyName.py

```diff
@@ -35,7 +35,7 @@ def check_key(self, key, path):
         if not isinstance(key, six.string_types):
             message = 'Mapping key ({0}) has to be a string.'
             matches.append(RuleMatch(path[:], message.format(key)))
-        elif not re.match('^[a-zA-Z0-9.-]{1,255}$', key):
+        elif not re.match('^[a-zA-Z0-9.-]{1,255}$', key) and key != 'Fn::Transform':
```

Comment:
Cool with this, but had a wilder general thought on https://github.com/aws-cloudformation/cfn-lint/issues/476

What if every violation on [`Transform`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-transform.html)  lines were ignored, regardless of individual rule logic?

Reply:
I was more filtering this out because we didn't flag this as a failure, then we started flagging it as a failure so wanted to bring it back to where it was.

Interesting.  It could be possible. We would want to evaluate the start and end mark of the finding is within the transform so if someone is doing a one line json document we don't filter out everything.  

As we continue to evaluate the need of credentials we may be able to test resolving the Transform as well.  Your option would be a great way to not have an issue if we can't resolve credentials.  We may be able to implement your idea fairly easily.  I may test it out today actually :)